src/solver/variable/state.cpp

```diff
@@ -254,263 +255,285 @@ void State::yearEndBuildFromThermalClusterIndex(const uint clusterAreaWideIndex,
+    assert(currentCluster);
+    assert(currentCluster->series);
+    assert(timeseriesIndex != NULL);
+
+    if (currentCluster->fixedCost > 0.)
+    {
+        dur = static_cast<uint>(
+        Math::Floor(currentCluster->startupCost / currentCluster->fixedCost));
+        if (dur > endHourForCurrentYear)
+            dur = endHourForCurrentYear;
+    }
+    else
+        dur = endHourForCurrentYear;
+    
+    for (h = startHourForCurrentYear; h < endHourForCurrentYear; ++h)
+    {
+        maxUnitNeeded = 0u;
+        ON_min[h] = 0u;
+        ON_max[h] = 0u;
 
-        assert(endHourForCurrentYear <= Variable::maxHoursInAYear);
-        assert(endHourForCurrentYear <= currentCluster->series->series.height);
-        assert(currentCluster);
-        assert(currentCluster->series);
-        assert(timeseriesIndex != NULL);
+        // Getting available production from cluster data
+        thermalClusterAvailableProduction = currentCluster->series->series[serieIndex][h];
 
-        if (currentCluster->fixedCost > 0.)
+        if (currentCluster->mustrun)
         {
-            dur = static_cast<uint>(
-              Math::Floor(currentCluster->startupCost / currentCluster->fixedCost));
-            if (dur > endHourForCurrentYear)
-                dur = endHourForCurrentYear;
+            // When the cluster is in must-run mode, the production value
+            // directly comes from the time-series
+            thermalClusterProduction
+            = thermalClusterAvailableProduction; // in mustrun, production==available
+                                                // production
         }
         else
-            dur = endHourForCurrentYear;
-
-        // min, and max unit ON calculation
-        for (h = startHourForCurrentYear; h < endHourForCurrentYear; ++h)
         {
-            maxUnitNeeded = 0u;
-            ON_min[h] = 0u;
-            ON_max[h] = 0u;
+            // otherwise from the solver results (most of the time)
+            thermalClusterProduction = thermalClusterProductionForYear[h];
+        }
 
-            // Getting available production from cluster data
-            thermalClusterAvailableProduction = currentCluster->series->series[serieIndex][h];
+        if (thermalClusterProduction > 0.)
+        {
+            thermalClusterOperatingCostForYear[h]
+            = (thermalClusterProduction * currentCluster->productionCost[h]);
 
-            if (currentCluster->mustrun)
+            switch (unitCommitmentMode)
             {
-                // When the cluster is in must-run mode, the production value
-                // directly comes from the time-series
-                thermalClusterProduction
-                  = thermalClusterAvailableProduction; // in mustrun, production==available
-                                                       // production
-            }
-            else
+            case Antares::Data::UnitCommitmentMode::ucFast:
```

Comment:
This `switch..case` is used to determine `ON_min[h]`. It would be nice to make it visible here with a function
```cpp
ON_min[h] = minimalClustersON(unitCommitmentMode, currentCluster, /* ... */);
```

Lines 334-370.